src/main/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitChecker.java

```diff
@@ -0,0 +1,153 @@
+                            expiration = rateLimit.getResetDate().getTime() + ENTROPY.nextInt(65536); // approx 1 min
+                            listener.getLogger().println(GitHubConsoleNote.create(System.currentTimeMillis(), String.format(
+                                    "GitHub API Usage: Current quota has %d remaining (%d over budget). Next quota of %d in %s. Sleeping until reset.",
+                                    rateLimit.remaining, ideal - rateLimit.remaining, rateLimit.limit,
+                                    Util.getTimeSpanString(rateLimitResetMillis)
+                            )));
+                        }
+                    } else {
+                        // work out how long until remaining == ideal + 0.1 * buffer (to give some spend)
+                        double targetFraction = (rateLimit.remaining - buffer * 1.1) / (rateLimit.limit - buffer - burst);
+                        expiration = rateLimit.getResetDate().getTime()
+                                - Math.max(0, (long) (targetFraction * MILLIS_PER_HOUR))
+                                + ENTROPY.nextInt(1000);
+                        listener.getLogger().println(GitHubConsoleNote.create(System.currentTimeMillis(), String.format(
+                                "GitHub API Usage: Current quota has %d remaining (%d over budget). Next quota of %d in %s. Sleeping for %s.",
+                                rateLimit.remaining, ideal - rateLimit.remaining, rateLimit.limit,
+                                Util.getTimeSpanString(rateLimitResetMillis),
+                                Util.getTimeSpanString(expiration - System.currentTimeMillis())
+                        )));
+                    }
+                    waitUntilRateLimit(listener, github, rateLimit, expiration);
+                }
+            }
+        }
+    },
+
+    /**
+     * Restrict GitHub API requests only when near or above rate limit.
+     */
+    ThrottleOnOver(Messages.ApiRateLimitChecker_ThrottleOnOver()) {
+        @Override
+        public void checkApiRateLimit(@NonNull TaskListener listener, GitHub github) throws IOException, InterruptedException {
+            boolean check = true;
+            while (check) {
+                GHRateLimit rateLimit = github.rateLimit();
+                // the buffer is how much we want to avoid using to cover unplanned over-use
+                int buffer = Math.max(15, rateLimit.limit / 20);
+                // check that we have at least our minimum buffer of remaining calls
+                if (rateLimit.remaining >= buffer) {
+                    break;
+                }
+                final long expiration = System.currentTimeMillis() + ENTROPY.nextInt(65536); // approx 1 min
+                listener.getLogger().println(GitHubConsoleNote.create(System.currentTimeMillis(), String.format(
+                        "GitHub API Usage: Current quota has %d remaining (%d over buffer). Next quota of %d due now. Sleeping for %s.",
```

Comment:
Is this log message correct? `Next quota of %d due now.` 
It's weird seeing that in logs and wondering why it's sleeping